add connect generator and modify class to accept array input

bmeverett opened the pull request:
Adds the Connect generator. Minor modifications to the class to accept an array as input.

Resolves #404

ErikSchierboom commented on exercises/connect/ConnectTest.cs:
```diff
@@ -1,124 +1,86 @@
+    public void An_empty_board_has_no_winner()
     {
-        return string.Join("\n", board.Select(x => x.Replace(" ", "")));
+        var board = new[] { ".....", ".....", ".....", ".....", "....." };
```
For the Connect case, it is quite important that the input strings are aligned nicely under each other, to allow users to see the pattern that needs to be connected. By default, the generator framework does not support this. but perhaps you can "borrow" from [this generator](https://github.com/exercism/csharp/pull/457/files#diff-2de73a257c3f6aa9810bdd1be46c7ab4R37)?

ErikSchierboom commented on generators/Exercises/Connect.cs:
```diff
@@ -0,0 +1,43 @@
+                canonicalDataCase.SetConstructorInputParameters("board");
+
+                //remove whitespace
+                var trimBoard = new List<string>();
```
Why trim the white space? The original implementation also doesn't trim it and this makes it look better IMHO. Could you remove the trimming?

bmeverett replied:
If I don't trim it the tests where failing because I think it's looping through every character and taking the spaces as an empty space. I could be wrong, I'll look into it.

bmeverett commented:
@ErikSchierboom I pushed up some changes. I aligned the arrays better, but I couldn't figure out how to get rid of the empty line at the end of the array. Besides that, hopefully I addressed your comments.

ErikSchierboom commented on exercises/connect/ConnectTest.cs:
```diff
@@ -1,124 +1,160 @@
     {
-        return string.Join("\n", board.Select(x => x.Replace(" ", "")));
+        var board = new [] 
+                        { 
```
Could you perhaps use less indentation? Something like this looks better IMHO:

```csharp
var board = new [] 
{ 
    ". . . . .",
    " . . . . .",
    "  . . . . .",
    "   . . . . .",
    "    . . . . ."
};
```

ErikSchierboom commented:
Merged. Thanks a lot. 🎉
